apps/framework-cli/src/cli/routines.rs

```diff
@@ -362,24 +366,32 @@ async fn manage_leadership_lock(
         client.check_and_renew_lock("leadership").await?
     };
 
-    if has_lock && is_new_acquisition {
-        info!("<RedisClient> Obtained leadership lock, performing leadership tasks");
-
-        let project_clone = project.clone();
-        tokio::spawn(async move {
-            if let Err(e) = leadership_tasks(project_clone).await {
-                error!("<RedisClient> Error executing leadership tasks: {}", e);
+    if has_lock {
+        if is_new_acquisition && !IS_RUNNING_LEADERSHIP_TASKS.load(Ordering::SeqCst) {
+            info!("<RedisClient> Obtained leadership lock, performing leadership tasks");
+
+            IS_RUNNING_LEADERSHIP_TASKS.store(true, Ordering::SeqCst);
```

Comment:
Do you set this back to false when the leadership task is lost?

Reply:
Yes updating PR